controllers/component_build_controller_pac.go

```diff
@@ -535,7 +536,26 @@ func (r *ComponentBuildReconciler) UnconfigureRepositoryForPaC(ctx context.Conte
 
 	isAppUsed := IsPaCApplicationConfigured(gitProvider, pacConfig)
 	if !isAppUsed {
-		if webhookTargetUrl != "" {
+		componentList := &appstudiov1alpha1.ComponentList{}
+		if err := r.Client.List(ctx, componentList, &client.ListOptions{Namespace: component.Namespace}); err != nil {
+			log.Error(err, "failed to list components")
+			return "", "", "", err
+		}
+		gitUrl := strings.TrimSuffix(strings.TrimSuffix(component.Spec.Source.GitSource.URL, ".git"), "/")
```

Comment:
nitpick: this line sorta separates operation on components list. I'd move it at the first line of the block (before working with the component list).